llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -1196,9 +1200,14 @@ bool VectorCombine::scalarizeLoadExtract(Instruction &I) {
   // Replace extracts with narrow scalar loads.
   for (User *U : LI->users()) {
     auto *EI = cast<ExtractElementInst>(U);
-    Builder.SetInsertPoint(EI);
-
     Value *Idx = EI->getOperand(1);
+
+    // Insert 'freeze' for poison indexes.
+    DenseMap<ExtractElementInst *, ScalarizationResult>::iterator It;
```

Comment:
`auto It = NeedFreeze.find(EI)`. Not need to spell out the iterator type, and I'm not sure why you did the assignment inside the if.

Reply:
Adapted to `auto It = NeedFreeze.find(EI)`. Thanks!